addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,635 @@
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.amazonechocontrol.internal.AccountConfiguration;
+import org.openhab.binding.amazonechocontrol.internal.Connection;
+import org.openhab.binding.amazonechocontrol.internal.ConnectionException;
+import org.openhab.binding.amazonechocontrol.internal.LoginServlet;
+import org.openhab.binding.amazonechocontrol.internal.StateStorage;
+import org.openhab.binding.amazonechocontrol.internal.discovery.AmazonEchoDiscovery;
+import org.openhab.binding.amazonechocontrol.internal.discovery.IAmazonEchoDiscovery;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonBluetoothStates;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonBluetoothStates.BluetoothState;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonDevices.Device;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonFeed;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonSmartHomeDevice;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Handles the connection to the amazon server.
+ *
+ * @author Michael Geramb - Initial Contribution
+ */
+public class AccountHandler extends BaseBridgeHandler implements IAmazonEchoDiscovery {
+
+    private final Logger logger = LoggerFactory.getLogger(AccountHandler.class);
+    private StateStorage stateStorage;
+    private AccountConfiguration config;
+    private Connection connection;
+    private List<EchoHandler> echoHandlers = new ArrayList<>();
+    private List<SmartHomeBaseHandler> smartHomeHandlers = new ArrayList<>();
+    private List<FlashBriefingProfileHandler> flashBriefingProfileHandlers = new ArrayList<>();
+    private Object synchronizeConnection = new Object();
+    private Map<String, Device> jsonSerialNumberDeviceMapping = new HashMap<>();
+    private ScheduledFuture<?> refreshJob;
+    private ScheduledFuture<?> refreshLogin;
+    private boolean updateSmartHomeDeviceList;
+    private boolean discoverFlashProfiles;
+    private boolean smartHodeDeviceListEnabled;
+    private String currentFlashBriefingJson = "";
+    private HttpService httpService;
+    private LoginServlet loginServlet;
+
+    public AccountHandler(@NonNull Bridge bridge, @NonNull HttpService httpService) {
+        super(bridge);
+        this.httpService = httpService;
+        stateStorage = new StateStorage(bridge);
+        AmazonEchoDiscovery.setHandlerExist();
+
+    }
+
+    @Override
+    public void initialize() {
+        start();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command '{}' received for channel '{}'", command, channelUID);
+
+        if (command instanceof RefreshType) {
+            refreshData();
+        }
+    }
+
+    public Device[] getLastKnownDevices() {
+        Map<String, Device> temp = jsonSerialNumberDeviceMapping;
```

Comment:
`temp` is not a proper name, please choose a better one :-)

Reply:
Ok, is it allowed, that I take the same name for a local variable and a member? I make the copy for multithreading reason. If it is ok, I would change it to  Map<String, Device> jsonSerialNumberDeviceMapping = this.jsonSerialNumberDeviceMapping;

Reply:
In this case you can just resolve it by returning a copy of the list this solves the next comment as well.

I’m not sure whether the result of values is a list which actually still points to the actual map, the doc block should telll you that, in that case you can return

Return new arraylist<>(mapping.values());

Reply:
Ok, all temp variables renamed